nexus_constructor/component.py

```diff
@@ -245,16 +247,48 @@ def depends_on(self, transformation: TransformationModel):
             )
             transformation.register_dependent(self)
 
+    def set_cylinder_shape(
+        self,
+        axis_direction: QVector3D = QVector3D(0.0, 0.0, 1.0),
+        height: float = 1.0,
+        radius: float = 1.0,
+        units: str = "m",
+    ) -> CylindricalGeometry:
+        """
+        Sets the shape of the component to be a cylinder
+        Overrides any existing shape
+        """
+        self._remove_shape()
+        validate_nonzero_qvector(axis_direction)
+        shape_group = self.file.create_nx_group(
+            "shape", "NXcylindrical_geometry", self.group
+        )
+        vertices = calculate_vertices(axis_direction, height, radius)
+        vertices_field = self.file.set_field_value(shape_group, "vertices", vertices)
+        self.file.set_field_value(shape_group, "cylinders", np.array([0, 1, 2]))
+        self.file.set_attribute_value(vertices_field, "units", units)
+        return CylindricalGeometry(self.file, shape_group)
 
-@attr.s
-class Component:
-    """DEPRECATED: Switching to use ComponentModel everywhere"""
-
-    nx_class = attr.ib(str)
-    name = attr.ib(str)
-    description = attr.ib(default="", type=str)
-    transform_parent = attr.ib(default=None, type=object)
-    dependent_transform = attr.ib(default=None, type=Transformation)
-    transforms = attr.ib(factory=list, type=List[Transformation])
-    geometry = attr.ib(default=None, type=Geometry)
-    pixel_data = attr.ib(default=None, type=PixelData)
+    def set_off_shape(self, loaded_geometry: OFFGeometry) -> OFFGeometryNexus:
+        """
+        Sets the shape of the component to be a mesh
+        Overrides any existing shape
+        """
+        self._remove_shape()
+        shape_group = self.file.create_nx_group("shape", "NXoff_geometry", self.group)
+        record_faces_in_file(self.file, shape_group, loaded_geometry.faces)
+        record_vertices_in_file(self.file, shape_group, loaded_geometry.vertices)
+        return OFFGeometryNexus(self.file, shape_group)
+
+    def get_shape(self) -> Optional[Union[OFFGeometry, CylindricalGeometry]]:
+        if "shape" in self.group:
```

Comment:
there are a few mentions of "shape" here. perhaps we should put this into a variable to avoid potential typos?